app/cli.f90

```diff
@@ -393,4 +396,56 @@ subroutine license(unit)
 end subroutine license
 
 
+subroutine list_funcs(unit)
+   integer, intent(in) :: unit
+
+   character(len=*), parameter :: func(*) = [character(len=32)::&
+      & 'am05', 'b1-lyp', 'b1-p', 'b1-pw', 'b1pw', 'b1pw91', 'b1p', 'b1p86', &
+      & 'b1lyp', 'b2-plyp', 'b2gp-plyp', 'b2plyp', 'b3-lyp', 'b3-p', 'b3-pw', &
+      & 'b3lyp', 'b3p', 'b3p86', 'b3pw', 'b3pw91', 'b6k', 'bh-lyp', 'bhpbe', &
+      & 'bmk', 'bp', 'bp86', 'bpbe', 'bpw', 'bwb6k', 'cam-b3lyp', 'dftb(3ob)', &
+      & 'dftb(matsci)', 'dftb(mio)', 'dftb(ob2)', 'dftb(pbc)', 'dsd-blyp', &
+      & 'dsd-pbe', 'dsd-pbep86', 'dsd-pbeb95', 'dsd-svwn', 'dodblyp', &
+      & 'dod-pbe', 'dod-pbep86', 'dod-pbeb95', 'dod-svwn', 'dsdblyp', &
+      & 'dsdpbe', 'dsdpbep86', 'dsdpbeb95', 'dsdsvwn', 'g-lyp', 'glyp', &
+      & 'hcth120', 'hcth407', 'hcthhyb', 'hf', 'hf-3c', 'hf-3cv', 'hf3c', &
+      & 'hf3cv', 'hiss', 'hse03', 'hse06', 'hse12', 'hse12s', 'hsesol', &
+      & 'lb94', 'lc-blyp', 'lc-dftb', 'lc-wpbe', 'lc-whpbe', 'lh07ssvwn', &
+      & 'lh07s-svwn', 'lh07tsvwn', 'lh07t-svwn', 'lh12ctssifpw92', &
+      & 'lh12ct-ssifpw92', 'lh12ctssirpw92', 'lh12ct-ssirpw92', 'lh14tcalpbe', &
+      & 'lh14t-calpbe', 'lh20t', 'lcwpbe', 'm05', 'm052x', 'm05-2x', 'm06', &
+      & 'm06-2x', 'm06l', 'm08hx', 'm08-hx', 'm11l', 'mn12l', 'mn12sx', &
+      & 'mn15', 'mn15l', 'mpw-lyp', 'mpw1-lyp', 'mpw1-pw', 'mpw1-pw91', &
+      & 'mpw1-pw', 'mpw1lyp', 'mpw1pw', 'mpw1pw91', 'mpw2plyp', 'mpwkcis1k', &
+      & 'mpwlyp', 'mpw-pw', 'mpwpw', 'mpwpw91', 'mpw1kcis', 'mpwb1k', 'n12', &
+      & 'n12sx', 'o-lyp', 'o3-lyp', 'o3lyp', 'olyp', 'omegab97', 'omegab97m', &
+      & 'opbe', 'otpss', 'pbe', 'pbe0', 'pbe02', 'pbe0-2', 'pbe0dh', &
+      & 'pbe0-dh', 'pbexalpha', 'pbeh1pbe', 'pbeh-3c', 'pbep86', 'pbepbeb95', &
+      & 'pbepbe', 'pbesol', 'pwp', 'pwp1', 'pw1-pw', 'pw1pw', 'pw1pw91', &
+      & 'pw6b95', 'pw86pbe', 'pw91', 'pw91p86', 'pwgga', 'revdsd-blyp', &
+      & 'revdsdblyp', 'revdsd-pbe', 'revdsdpbe', 'revdsd-pbepbe', &
+      & 'revdsdpbepbe', 'revdsdpbep86', 'revdsdpbeb95', 'revdsd-svwn', &
+      & 'revdod-pbep86', 'revdodpbep86', 'revdod-pbeb95', 'revdodpbepb95', &
+      & 'revdod-svwn', 'revpbe', 'revpbe0', 'revpbe0-dh', 'revpbe38', &
+      & 'revtpss', 'revtpss0', 'r2scan', 'r²scan', 'r2scan0', 'r²scan0', &
+      & 'r2scan50', 'r²scan50', 'r2scanh', 'r²scanh', 'rscan', 'scan', 'ssb', &
+      & 'tauhctc', 'tauhcthhyb', 'thcth', 'tpss', 'tpss0', 'tpssh', 'wb97', &
+      & 'wb97m', 'ωb97', 'ωb97m', 'x-lyp', 'x3-lyp', 'x3lyp']
+
+   character(len=1) :: current_letter = ''
+   integer :: i
+   
+   write(unit, '(a)') &
+      & 'The following functionals are available (keywords for --func):'
+  
+   do i = 1, size(func)
+      if (func(i)(1:1) /= current_letter) then
```

Comment:
This doesn't seem right, better check for the internal ID to do the grouping.

Reply:
I did not want to group, I just wanted to have some alphabetical separation. This works, because the array is already alphabetically sorted.